go/worker/compute/merge/committee/node.go

```diff
@@ -746,6 +748,11 @@ func (n *Node) worker() {
 				n.commonNode.CrossNode.Lock()
 				defer n.commonNode.CrossNode.Unlock()
 
+				if state, ok := n.state.(StateProcessingMerge); !ok || state.doneCh != mergeDoneCh {
+					n.logger.Warn("state snapshot changed unexpectedly, aborting")
```

Comment:
No need to warn here, maybe log at debug level or just remove as this is expected to happen.